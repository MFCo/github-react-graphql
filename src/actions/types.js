import keyMirror from 'key-mirror';

export default keyMirror({
  UPDATE_USER: null,
  UPDATE_REPOSITORIES: null,
  ADD_REPOSITORIES: null,
  UPDATE_DIRECTORY: null,
  UPDATE_REPOSITORY: null
});