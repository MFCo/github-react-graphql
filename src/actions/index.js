import ACTIONS from './types';
import { createActions } from 'redux-actions';

export default createActions(...Object.keys(ACTIONS));