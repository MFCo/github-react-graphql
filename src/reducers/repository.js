import { handleActions } from 'redux-actions';

const defaultState = {
  name: '',
  owner: '',
  description: '',
  homepage: '',
  id: '',
  starred: false,
  watching: false,
};

export default handleActions({
  UPDATE_REPOSITORY: (state, { payload: { newName, newOwner, newDescription, newHomepage, newId, newStarred, newWatching } }) => {
    return {
      name: newName,
      owner: newOwner,
      description: newDescription,
      homepage: newHomepage,
      id: newId,
      starred: newStarred,
      watching: newWatching
    }
  },
  STAR_CHANGE: (state, { payload: { newStarred } }) => {
    return {
      ...state,
      starred: newStarred
    }
  },
  WATCH_CHANGE: (state, { payload: { newWatching } }) => {
    return {
      ...state,
      watching: newWatching
    }
  }
}, defaultState);