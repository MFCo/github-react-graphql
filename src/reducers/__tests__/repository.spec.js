import repository from '../repository';
import actions from '../../actions';

it('should return the initial state', () => {
  expect(repository(undefined, {})).toMatchSnapshot()
})

it('should handle UPDATE_USER', () => {
  expect(repository(undefined, actions.updateRepository({
    newName: 'Jhon-Doe-Repo',
    newOwner: 'JDoe',
    newDescription: 'hi!',
    newHomepage: 'com.ar.jhon'
  }))).toMatchSnapshot()
})