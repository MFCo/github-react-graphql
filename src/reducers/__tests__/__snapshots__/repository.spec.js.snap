// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle STAR_CHANGE 1`] = `
Object {
  "description": "",
  "homepage": "",
  "id": "",
  "name": "",
  "owner": "",
  "starred": false,
  "watching": false,
}
`;

exports[`should handle UPDATE_REPOSITORY 1`] = `
Object {
  "description": "hi!",
  "homepage": "com.ar.jhon",
  "id": "asdaksdlksd==",
  "name": "Jhon-Doe-Repo",
  "owner": "JDoe",
  "starred": false,
  "watching": undefined,
}
`;

exports[`should handle WATCH_CHANGE 1`] = `
Object {
  "description": "",
  "homepage": "",
  "id": "",
  "name": "",
  "owner": "",
  "starred": false,
  "watching": false,
}
`;

exports[`should return the initial state 1`] = `
Object {
  "description": "",
  "homepage": "",
  "id": "",
  "name": "",
  "owner": "",
  "starred": false,
  "watching": false,
}
`;
